Source/effects.cpp

```diff
@@ -1011,83 +1011,54 @@ void __fastcall PlaySFX(int psfx)
 	PlaySFX_priv(&sgSFX[v1], 0, 0, 0);
 }
 
-void __fastcall PlaySFX_priv(TSFX *pSFX, char loc, int x, int y)
+void __fastcall PlaySFX_priv(TSFX *pSFX, BOOL loc, int x, int y)
 {
-	int v4; // edi
-	TSFX *v5; // esi
-	TSnd *v6; // ecx
-	//int v7; // eax
-	TSnd *v8; // ecx
-	int volume_delta; // [esp+Ch] [ebp-8h]
-	int pan; // [esp+10h] [ebp-4h]
+	if ( plr[myplr].pLvlLoad && gbMaxPlayers != 1 ) {
+		return;
+	}
+	if ( !gbSndInited || !gbSoundOn || gbBufferMsgs ) {
+		return;
+	}
 
-	v4 = loc;
-	v5 = pSFX;
-	if ( !plr[myplr].pLvlLoad || gbMaxPlayers == 1 )
-	{
-		if ( gbSndInited )
-		{
-			if ( gbSoundOn )
-			{
-				if ( !gbBufferMsgs )
-				{
-					if ( pSFX->bFlags & 3 || (v6 = pSFX->pSnd) == 0 || !snd_playing(v6) )
-					{
-						pan = 0;
-						volume_delta = 0;
-						if ( !v4 || calc_snd_position(x, y, &volume_delta, &pan) )
-						{
-							if ( v5->bFlags & 1 )
-							{
-								stream_play(v5, volume_delta, pan);
-							}
-							else
-							{
-								if ( !v5->pSnd )
-									v5->pSnd = sound_file_load(v5->pszName);
-								v8 = v5->pSnd;
-								if ( v8 )
-									snd_play_snd(v8, volume_delta, pan);
-							}
-						}
-					}
-				}
-			}
-		}
+	if ( !(pSFX->bFlags & 3) && pSFX->pSnd != 0 && snd_playing(pSFX->pSnd) ) {

```

Comment:
We also have some of the `pSFX->bFlags` enums from assert strings of the debug release :)

```c
pSFX->bFlags & sfx_STREAM
```

(only `sfx_STREAM` is known for sure, the rest are best effort guesses)
```c
typedef enum {
	sfx_STREAM   = 0x01,
	sfx_MISC     = 0x02,
	sfx_UI       = 0x04,
	sfx_08       = 0x08,
	sfx_ROGUE    = 0x10,
	sfx_WARRIOR  = 0x20,
	sfx_SORCEROR = 0x40,
	sfx_LOADED   = 0x80,
} sfx_flag;
```